Modules/_io/_iomodule.c

```diff
@@ -386,16 +386,20 @@ _io_open_impl(PyObject *module, PyObject *file, const char *mode,
         return result;
     }
 
+    _PyIO_State *state = IO_STATE();
```

Comment:
FYI: when the ref leak run completes, I'll remove this and move the state in line 318 to the function scope.